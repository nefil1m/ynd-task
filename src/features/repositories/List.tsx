import { ReactElement } from 'react';
import { Repository } from './repositoriesSlice';
import styles from './List.module.scss';

const StarIcon = () => (
  <svg viewBox="0 0 512 512">
    <polygon strokeWidth="37.6152" strokeLinecap="round" strokeLinejoin="round" strokeMiterlimit="10" points="259.216,29.942 330.27,173.919 489.16,197.007 374.185,309.08 401.33,467.31 259.216,392.612 117.104,467.31 144.25,309.08   29.274,197.007 188.165,173.919" />
  </svg>
);

export const RepositoryCard = ({
  name,
  description,
  stargazersCount,
}: Repository): ReactElement => (
  <article className={styles.repositoryCard} data-testid="repository-card">
    <header className={styles.repositoryCardHeader}>
      <span className={styles.repositoryCardTitle}>{name}</span>
      <span className={styles.repositoryCardStars}>
        {stargazersCount}
        <StarIcon />
      </span>
    </header>
    <p>
      {description?.length > 75
        ? `${description.substring(0, 75)}...`
        : description}
    </p>
  </article>
);

interface RepositoryListProps {
  repositories: Repository[];
}

const RepositoryList = ({ repositories }: RepositoryListProps): ReactElement => (
  <ul className={styles.list}>
    {repositories.map(({
      name,
      id,
      stargazersCount,
      description,
    }) => (
      <li key={id}>
        <RepositoryCard
          id={id}
          name={name}
          description={description}
          stargazersCount={stargazersCount}
        />
      </li>
    ))}
  </ul>
);

export default RepositoryList;
