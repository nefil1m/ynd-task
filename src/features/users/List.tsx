import { ReactElement } from 'react';
import { Accordion, AccordionContent, AccordionToggle } from '../../components/Accordion/Accordion';
import Repositories from '../repositories';
import { User } from './usersSlice';
import styles from './List.module.scss';

const UserAccordion = ({ login }: User) => (
  <Accordion>
    <AccordionToggle>
      {login}
    </AccordionToggle>
    <AccordionContent>
      <Repositories username={login} />
    </AccordionContent>
  </Accordion>
);

interface UsersListProps {
  users: User[];
}

const UsersList = ({ users }: UsersListProps): ReactElement => (
  <div data-testid="users-list">
    <ul className={styles.list}>
      {users.map(({ id, login }) => (
        <li key={id}>
          <UserAccordion
            id={id}
            login={login}
          />
        </li>
      ))}
    </ul>
  </div>
);

export default UsersList;
