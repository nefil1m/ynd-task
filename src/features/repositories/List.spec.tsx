import { render } from '@testing-library/react';
import { RepositoryCard } from './List';

describe('repository card', () => {
  it('should display empty state', async () => {
    const { asFragment } = render(
      <RepositoryCard
        id="repo1"
        name="Hello world"
        description="Description"
        stargazersCount="4"
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
