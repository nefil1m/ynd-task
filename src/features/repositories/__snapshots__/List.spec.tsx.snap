// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repository card should display empty state 1`] = `
<DocumentFragment>
  <article
    class="repositoryCard"
    data-testid="repository-card"
  >
    <header
      class="repositoryCardHeader"
    >
      <span
        class="repositoryCardTitle"
      >
        Hello world
      </span>
      <span
        class="repositoryCardStars"
      >
        4
        <svg
          viewBox="0 0 512 512"
        >
          <polygon
            points="259.216,29.942 330.27,173.919 489.16,197.007 374.185,309.08 401.33,467.31 259.216,392.612 117.104,467.31 144.25,309.08   29.274,197.007 188.165,173.919"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-miterlimit="10"
            stroke-width="37.6152"
          />
        </svg>
      </span>
    </header>
    <p>
      Description
    </p>
  </article>
</DocumentFragment>
`;
